source/daplink/interface/main.c

```diff
@@ -363,6 +366,31 @@ void main_task(void * arg)
                 reset_pressed = 0;
             }
 
+#ifdef PBON_BUTTON
+            // handle PBON pressed
+            if(gpio_get_pbon_btn())
+            {
+                if(power_on)
+                {
+                    // Loop till PBON is pressed
+                    while (gpio_get_pbon_btn()) {;}
+                    // Power button released when target was running
+                    target_set_state(SHUTDOWN);
+                    power_on = 0;
+                    reset_pressed = 1;
+                }
+                else
+                {
+                    // Loop till PBON is pressed
+                    while (gpio_get_pbon_btn()) {;}
+                    // Power button released when target was already powered off
+                    target_set_state(POWER_ON);
+                    power_on = 1;
+                    reset_pressed = 0;
+                }
+            }
+#endif
+
```

Comment:
Is this logic really necessary? Daplink should not care about powering off and on the target in my opinion

Reply:
There is power on/off sequence which needs to be executed, hence the logic

Reply:
there is power on/off sequence that needs to be executed, hence the logic as PBON button is just another button connected to lpc11u35

Reply:
We do not allow target-specific code in common files. You'll have to add a clean hook mechanism so you can implement the power button support in your board file.

Although as a user, I'd much prefer the power button was not required. It took a while for me to understand I had to press the button to power up the target when I first got my Musca-A1 board. At the very least, it should automatically power up the target if the CMSIS-DAP debug connection is enabled.